### PR TITLE
feat: add docker container

### PR DESCRIPTION
**Pull request by Trusted97**

### CODE REVIEW

**Review by antonioturdo — APPROVED** [review]

**Review by JellyBellyDev — COMMENTED** [review]
Can't we use docker images that already have composer inside them?
This way we avoid using `build` and we avoid `Dockerfile`...
WDYT?

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: '3.9'
+
+services:
+  php-8.0: &php
```
Comment (JellyBellyDev) [inline docker-compose.yaml:4]: wouldn't it be better `php80`? so as to make it easier to use from the cli with `docker-compose exec php80 sh`?

Comment (Trusted97) [reply docker-compose.yaml:4]: Ok i'll change it in next commit

Comment (Trusted97) [reply docker-compose.yaml]: Done @JellyBellyDev

**Comment by Trusted97** [comment]
> Can't we use docker images that already have composer inside them? This way we avoid using `build` and we avoid `Dockerfile`... WDYT?

@JellyBellyDev 

I haven't found any official images that have PHP and Composer installed together.

Furthermore, the idea of ​​dividing and organizing docker like this is an idea that I have already seen used successfully in other libraries and could allow us to easily make changes in the future. WDYT?

**Comment by JellyBellyDev** [comment]
> Furthermore, the idea of ​​dividing and organizing docker like this is an idea that I have already seen used successfully in other libraries and could allow us to easily make changes in the future. WDYT?

ok, that's okay!

**Comment by JellyBellyDev** [comment]
@Trusted97 wdyt to update CONTRIBUTING.md to use docker-compose?

**Comment by Trusted97** [comment]
@JellyBellyDev I've updated the CONTRIBUTING.md file WDYT?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,6 +10,28 @@
...
+
+```console
+docker-compose build --parallel
+```
+
+Now you can run commands needed to work on the project. For example, say you want to install the dependencies on PHP 8.0:
+
+```console
+docker-compose run php-80 composer install
+```
```
Comment (JellyBellyDev) [inline CONTRIBUTING.md:34]: we wolud to ad also che example to enter in container?

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,33 +1,23 @@
...
-    <filter>
-        <whitelist>
-            <directory suffix=".php">src</directory>
-        </whitelist>
-    </filter>
-
-    <logging>
-        <log type="coverage-html" target="coverage/" />
-    </logging>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd" backupGlobals="false" colors="true" bootstrap="vendor/autoload.php" cacheResult="false" convertDeprecationsToExceptions="false">
```
Comment (JellyBellyDev) [inline phpunit.xml.dist:3]: warning: https://github.com/immobiliare/braze-php-sdk/blob/main/composer.json#L24 
it's possibile to use also the phpunit 8.5!

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]

File: phpunit.xml.dist (hunk 3 above)
Comment (Trusted97) [reply phpunit.xml.dist]: Updated! @JellyBellyDev

File: CONTRIBUTING.md (hunk 2 above)
Comment (Trusted97) [reply CONTRIBUTING.md:34]: Updated @Trusted97

**Review by JellyBellyDev — APPROVED** [review]
LGTM 👍

**Pull request merged**